### ISSUE TITLE
FL0035: False positive with Active Patterns

### ISSUE THREAD

**Issue by Evangelink** [fsprojects/FSharpLint#484]
### Description

Composition cannot be applied (or I couldn't figure out how to) for Active Patterns so `FL0035` should not be raised.

### Repro steps

```fsharp
let (|Something|_|) (l: int list option) =
	l |> Option.map (fun x -> Something (System.Collections.ObjectModel.ObservableCollection x)) // FL0035 FP
```

### Expected behavior

No issue raised.

### Actual behavior

FL0035 is raised.

### Known workarounds

Don't know.

### Related information

* Operating system: Windows 10
* Branch: F# Lint 0.5.2
* .NET Runtime, CoreCLR or Mono Version: .net 48
